### ISSUE TITLE
🖼️Support iframes

### ISSUE THREAD

**Issue by jperl** [qawolf/qawolf#279]
Name page variables as part of this, since we will need to track variables.

**Comment by jperl** [comment]
If you want this please comment or add a reaction, so we know to prioritize it.

**Comment by jperl** [comment]
In this upcoming release we are switching to native playwright selectors. We should consider how this relates to https://github.com/microsoft/playwright/issues/152

**Comment by jperl** [comment]
This ticket is just waiting on https://github.com/microsoft/playwright/issues/152

**Comment by dohomi** [comment]
would be great to test storybook components, thanks for adding it

**Comment by jperl** [comment]
I will definitely test this works with storybook, thanks for the suggestion

**Comment by aldeed** [comment]
@jperl Initial investigation/planning...

I created a new sandbox page with iframes to determine what's currently not working. In a nutshell, looks like it detects actions within iframes, but it doesn't realize they're in an iframe so it generates `page.click`, etc. rather than `frame.click`.

Example current generated test:

```js
await page.goto("http://localhost:5000/iframes");
await page.click('[data-qa="html-text-input"]');
await page.fill('[data-qa="html-text-input"]', "foo");
await page.press('[data-qa="html-text-input"]', "Tab");
```

And I think where we want to end up is this:

```js
await page.goto("http://localhost:5000/iframes");

const frame1Element = await page.$('[data-qa="second"]'); // iframe element selector
const frame1 = await elementHandle.contentFrame();

await frame1.click('[data-qa="html-text-input"]');
await frame1.fill('[data-qa="html-text-input"]', "foo");
await frame1.press('[data-qa="html-text-input"]', "Tab");
```

Does this seem right? When I manually update the test to this, it works.

Questions:
1. `frameNElement` and `frameN` are ok for variable names?
2. In OP you say "remove" `qawolf.waitForPage`. Why and where is that needed?
3. It appears that playwright auto-waits-for within iframe same as on main page. If you don't think that's true, let me know.
4. I assume we can reuse existing selector build fn, with same penalties, for the iframe element selector.
5. Do we care about nested iframes for this initial support?
6. For verifying storybook support, do you want me to add storybook and some stories to the `sandbox` project?

**Comment by jperl** [comment]
1. I would prefer it to happen on one line if possible.  

`const frame1 = await (await page.$('[data-qa="second"]')).contentFrame();`

Or have a helper that does that: `await qawolf.contentFrame('[data-qa="second"]')`?

2. I guess that's bad wording. The goal is to have separate variables rather than remove `qawolf.waitForPage`. Right now when you do actions on a popup page we reassign the page variable `page =  await qawolf.waitForPage`, whereas it would be better to have different page variables. `const page1 = await qawolf.waitForPage` & `const page2 = await qawolf.waitForPage`.

3. Yes it should.

4. Yes.

5. No one layer is fine.

6. No just manually check it. However please do add iframe examples/tests to the sandbox.

**Comment by aldeed** [comment]
That all makes sense. I'll give it a go.

**Comment by jperl** [comment]
This is released in 1.3.3. Thanks @aldeed!
